src/deep_learning_container.py

```diff
@@ -308,14 +308,17 @@ def main():
 
     logging.basicConfig(level=logging.ERROR)
     token = None
+    instance_id = None
+    region = None
+    token = get_imdsv2_token()
+    if token:
+        instance_id = _retrieve_instance_id(token)
 
-    instance_id = _retrieve_instance_id()
     if instance_id:
```

Comment:
this condition can be completly avoided by adding `region = _retrieve_instance_region(token)` to the `if token:` condition

Reply:
I think both functions _retrieve_instance_id() and _retrieve_instance_region should be able to handle the case where token has a value or is None|undefined. Then, there would be no need need for any of the conditions, and all you will have is the code below.
`token = get_imdsv2_token()`
`instance_id = _retrieve_instance_id(token)`
`region = _retrieve_instance_region(token)`